quattor/types/hardware.pan

```diff
@@ -20,9 +20,12 @@ type structure_ram = {
 }
 type structure_cpu = {
     include structure_annotation
-    "speed"  : long # "CPU clock speed in MHz"
-    # Number of cores on each CPU chip, defaults to 1.
+    @{ desc = CPU clock speed in MHz }
+    "speed"  : long
+    @{ desc = Number of cores on each CPU chip }
     "cores" : long(1..) = 1
+    @{ desc = Number of execution threads on each CPU chip }
```

Comment:
Ability to generate configuration for batch systems/virtualisation platforms based on threads.
A hyperthreading boolean is not enough (core:thread ratio is not always 1:2) and as you mentioned in #94, some platforms even allow the ratio to be defined.

My meaning was per physical package (because it's easy data to get hold of) but threads per core would be straightforward enough as well.